### ISSUE TITLE
Documentation: Using PnP.PowerShell in Remote Containers

### ISSUE THREAD

**Issue by shurick81** [pnp/powershell#2003]
**Is your feature request related to a problem? Please describe.**
Not a big problem, but when I set up a new project that is using PnP.PowerShell, if I want to use Visual Studio remote containers, I need to come up with a new text of devcontainer.json. There seems no good description of the procedure and devcontainer.json to use.

**Describe the solution you'd like**
It would be probably great to have an article, similar to https://pnp.github.io/powershell/articles/azurecloudshell.html, called something like "Using PnP PowerShell in Visual Studio Remote Containers"

**Describe alternatives you've considered**
Someone could post it in their blogs, gists etc.

**Additional context**

**Comment by CallumCrowley** [comment]
@shurick81 Please use the Bug Report template when reporting bugs. You can do this by clicking New Issue > Get Started (next to Bug Report).

**Comment by veronicageek** [comment]
@CallumCrowley - This is not a bug, and @KoenZomers already added the documentation label.

**Comment by CallumCrowley** [comment]
@veronicageek apologies, I'm not sure how I missed that! 😆

**Comment by gautamdsheth** [comment]
hi @shurick81 , any update on this one as well ? Would be great to have this in our docs :)

**Comment by shurick81** [comment]
For me it would be nice to contribute consecutively, after #2008

**Comment by gautamdsheth** [comment]
hi @shurick81, you are our docker expert,so wanted your opinion :)

Would it make sense to have `devcontainer.json` kinda thing for PnP PowerShell ? If possible ?
I looked at the CLI M365 code, and they are using it as well. Check this --> 

https://github.com/pnp/cli-microsoft365/blob/main/.devcontainer/devcontainer.json

Is it possible to have something like this for PnP PowerShell ?

**Comment by shurick81** [comment]
Yes, I think we should also have something like this. For example, here's the docu that  I made for the SharePoint Framework: https://github.com/shurick81/docker-spfx/blob/feature/Remote-Containers-Docu/DevelopmentContainers.md

**Comment by shurick81** [comment]
Or maybe you mean using dev containers for the those who want to contribute to PnP.PowerShell? Yes, that is another improvement that I would suggest. If so, let's create a separate issue for that?

**Comment by gautamdsheth** [comment]
Both? If possible? Can you create them as enhancements? Sorry, am not even sure what to write for this. Would be a great improvement I think

**Comment by shurick81** [comment]
@gautamdsheth when I wanted to register this new enhancement suggestion, I stumbled into the fact that I know very little about how does it usually works for contributors to implement changes in the PnP.PowerShell module: what environment they need for building and debugging. Would you recommend any person to discuss this with?

**Comment by shurick81** [comment]
Hi @gautamdsheth, @veronicageek, could someone help me show some practices of contributing to new versions of this module? Then I could register a new issue for documenting how to use development containers.

**Comment by gautamdsheth** [comment]
hi @shurick81 , can you elaborate more , sorry not getting this. The repo is totally open-source and we welcome all contributions. We dont have any such defined practices because every person who wants to contribute is different and uses different environment.

I use Visual Studio, where after cloning the repo, I just need to do F5.
Erwin/Koen prefer VS code for development and Veronique uses Mac.
Other contributors may use different stuff as well.

The code works wherever PowerShell works :) 
For this issue/feature, maybe just send a PR whenever you are ready. You can showcase whatever you have used for containers, it is totally fine as long as it works :) 
Let me know if you need more info.
